dialog/input.go

```diff
@@ -0,0 +1,87 @@
+package dialog
+
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+// InputDialog is a variation of a dialog which prompts the user to enter some
+// text.
+type InputDialog struct {
+	*dialog
+
+	onConfirm func(string)
+
+	entry *widget.Entry
+
+	confirmButton *widget.Button
+}
+
+// SetText changes the current text value of the input dialog, this can
+// be useful for setting a default value.
+func (i *InputDialog) SetText(s string) {
+	i.entry.SetText(s)
+}
+
+// GetText returns the current text value entered in the input dialog.
```

Comment:
This access is not something we have done with other dialogs, can you share why it was added?

Reply:
The rationale was that if the user cancelled, you might still want access to the text? If that's stylistically inconsistent, it can be removed. I think the use case is probably pretty niche.

Reply:
Aha! I think the other ones call the callback on cancel with a nil value.
If you look at the confirm dialog I am pretty sure how that works.
If people use the ShowxxxDialog() function there is no reference to get the text after cancelling without callback so that's probably why we took that approach.

Reply:
So how would you handle that in this case? I kinda feel like there should be a separate callback for if the dialog is cancelled, since presumably the one that receives the value entered cannot do it's job without having a proper value.

Alternatively, I could have it take a `bool` argument that indicates if the user pressed cancel or confirm?

I don't like the idea of just calling it with the empty string, since in some cases that might be a legitimate input.

Reply:
I agree that we could do better to separate confirm vs dismiss callbacks.
However I also think that it should be consistent.
To fix the others is the aim but this is a breaking (2.0) change.

There is an OnClosed() callback that the developer can use in the meantime - this should be called irrespective of confirm or cancel. I guess given that we could have the input callback (in this case - with clear documentation) only call on confirm might just be ok.

Reply:
I went ahead and implemented this in 11d2c35b, just in case it might be useful for someone.